scripts/docs/check-broken-links.mjs

```diff
@@ -0,0 +1,168 @@
+import { readdir, readFile, stat } from 'node:fs/promises';
+import { dirname, relative, resolve } from 'node:path';
+
+import inspectUrls from '@jsdevtools/rehype-url-inspector';
+import slug from 'rehype-slug';
+import stringify from 'rehype-stringify';
+import gfm from 'remark-gfm';
+import parse from 'remark-parse';
+import rehype from 'remark-rehype';
+import { unified } from 'unified';
+
+const links = [];
+
+const getMarkdownFiles = async (directory) => {
+  const files = await readdir(directory, { withFileTypes: true });
+  const markdownFiles = [];
+
+  for await (const file of files) {
+    const fullPath = resolve(directory, file.name);
+
+    if (file.isDirectory()) {
+      const nestedFiles = await getMarkdownFiles(fullPath);
+      markdownFiles.push(...nestedFiles);
+    } else if (file.isFile() && file.name.endsWith('.md')) {
+      markdownFiles.push(fullPath);
+    }
+  }
+
+  return markdownFiles;
+};
+
+const logBrokenLink = ({ directory, filePath, position, url }) => {
+  const { line, column } = position;
+  links.push(`${directory}/${relative(directory, filePath)}:${line}:${column} - ${url}`);
+};
+
+const checkAnchorInFile = async ({ anchor, directory, filePath, originalUrl, position, sourceFile }) => {
+  const content = await readFile(filePath, 'utf8');
+  const u = await unified().use(gfm).use(parse).use(rehype).use(slug).use(stringify);
+  const result = await u.run(u.parse(content));
+
+  const anchorExists = result.children.some((i) => i.properties.id === anchor);
```

Comment:
This fixes the [error](https://github.com/romcal/romcal/actions/runs/10189573388/job/28187873100?pr=586#step:9:21) in the CI that `id` is not defined. Not all items have an `id`.

```suggestion
  const anchorExists = result.children.some((i) => i.properties?.id === anchor);
```

Reply:
there's other issues that i fixed on a dev machine that's waiting for a new cpu cooler